src/ansys/fluent/core/services/interceptors.py

```diff
@@ -61,6 +61,38 @@ def intercept_unary_unary(
         return self._intercept_call(continuation, client_call_details, request)
 
 
+class ErrorStateInterceptor(grpc.UnaryUnaryClientInterceptor):
+    """Interceptor class to trace gRPC calls."""
+
+    def __init__(self, fluent_error_state):
+        """__init__ method of ErrorStateInterceptor class."""
+        super().__init__()
+        self._fluent_error_state = fluent_error_state
+
+    def _intercept_call(
+        self,
+        continuation: Any,
+        client_call_details: grpc.ClientCallDetails,
+        request: Any,
+    ):
+        if self._fluent_error_state == "fatal":
+            details = self._fluent_error_state.details
+            self._fluent_error_state.clear()
```

Comment:
Is it better to clear the state in events_streaming.py in an else block of `if event_name == "errorevent"`?